pysph/base/opencl.py

```diff
@@ -35,6 +42,31 @@ def set_queue(q):
     _queue = q
 
 
+def profile(name, event):
+    global _profile_info
+    event.wait()
+    time = (event.profile.end - event.profile.start) * 1e-9
+    if name in _profile_info:
+        _profile_info[name] += time
+    else:
+        _profile_info[name] = time
+
+
+def profile_info():
+    global _profile_info
+    _profile_info = sorted(_profile_info.iteritems(), key=itemgetter(1),
+                           reverse=True)
+    print_profile(_profile_info)
+
+
+def print_profile(info):
+    if len(info) == 0:
+        return
```

Comment:
Instead print "no profile information available." or so, as you are promising to print something...